internal/elasticsearch/service_monitor.go

```diff
@@ -73,3 +98,47 @@ func (er *ElasticsearchRequest) CreateOrUpdateServiceMonitors() error {
 
 	return nil
 }
+
+func (er *ElasticsearchRequest) getServiceAccountTokenSecretName() (string, error) {
+	key := client.ObjectKey{
+		Name:      serviceMonitorServiceAccountName(er.cluster.Name),
+		Namespace: er.cluster.Namespace,
+	}
+
+	sa, err := serviceaccount.Get(context.TODO(), er.client, key)
+	if err != nil {
+		return "", kverrors.Wrap(err, "failed to get elasticsearch-metrics serviceacount",
+			"cluster", er.cluster.Name,
+			"namespace", er.cluster.Namespace,
+			"key", key,
+		)
+	}
+
+	var tokenSecret string
+	for _, oref := range sa.Secrets {
```

Comment:
This code will probably not work correctly on OCP 4.11 as ServiceAccounts do not get a Secret containing the token by default anymore on Kubernetes 1.24.

Reply:
Good catch, I have update the PR to create a ServiceAccountToken Secret manually. PTAL

Reply:
This works for me on 4.11.